java/feathub-udf/flink-udf/src/main/java/com/alibaba/feathub/flink/udf/processfunction/SlidingWindowKeyedProcessFunction.java

```diff
@@ -189,8 +201,11 @@ public void onTimer(
                     break;
                 }
                 Row curRow = state.timestampToRow.get(rowTime);
-                descriptor.aggFunc.retract(
-                        accumulatorState, curRow.getField(descriptor.inFieldName));
+                accumulatorState =
+                        aggFuncMap
+                                .get(descriptor.outFieldName)
+                                .retractAccumulator(
+                                        accumulatorState, curRow.getField(descriptor.outFieldName));
```

Comment:
Since we are not setting the `accumulatorState` back to `state.getAccumulatorStates()`, does this mean the correctness depends on `AggFunc#retractAccumulator` and `AggFunc#merge` always mutating the first aggregator parameter?

Should we update the signature of these two functions accordingly?